### PR TITLE
Don't show undef

### PR DESCRIPTION
**Pull request by peterschussheim**
Associated Issue: #2691 

### Summary of Changes

 

- [x] Don't show undefined values of variables in current scope when debugger is paused
- [x] Add selected-token class to popoverTarget in `componentDidMount`
- [x] Remove selected-token class from popoverTarget in `componentWillUnmount`

### Test Plan
- [x] When paused in the debugger, hovering over an undefined variable, does not add highlighting class or the popup preview.

### CODE REVIEW

File: src/utils/editor/expression.js [hunk 1]
```diff
@@ -64,6 +64,11 @@ export function previewExpression({
   }
 
   if (variables.has(tokenText)) {
+    let variableKey = variables.get(tokenText);
+    if (!get(variableKey, "contents.value.type") == "undefined") {
+      return null;
+    }
+
```
Comment (jasonLaster) [inline src/utils/editor/expression.js]: the next step is to look at the expression and see if its value is undefined too :)

don't worry, we'll be cleaning this code up soon :P

File: src/utils/editor/expression.js [hunk 2]
```diff
@@ -64,6 +64,11 @@ export function previewExpression({
   }
 
   if (variables.has(tokenText)) {
+    let variableKey = variables.get(tokenText);
```
Comment (jasonLaster) [inline src/utils/editor/expression.js]: I think we should try writing a test for this: 

here is the test file: https://github.com/peterschussheim/debugger.html/blob/dont-show-undef-latest/src/utils/tests/expression.js#L5-L31

Comment (peterschussheim) [inline src/utils/editor/expression.js]: OK, I will work on this today.

**Comment by jasonLaster** [comment]
Thanks @peterschussheim! I'm really impressed by your persistence. It was great to pair with you wednesday and get to the bottom of it

**Pull request merged**